examples/cpp/fill-grid-pp2hadron.cpp

```diff
@@ -0,0 +1,245 @@
+    for (int i = -3; i <= 3; ++i) {
+        if (i == 0) continue;
+        pids.insert(pids.end(), {21, 21, i});
+        factors.push_back(1.0);
+    }
+    pineappl_channels_add(channels, pids.size() / nb_convolutions, pids.data(),
+        factors.data());
+
+    // specify the perturbative orders that will be filled into the grid
+    // orders specifies the power of the tuple `orders = (αs, α, lR, lF, lD)`
+    // in this example, we only fill the LO QCD
+    std::vector<uint8_t> orders = {1, 0, 0, 0, 0};
+
+    // bin limits of the final-state hadron transverse momentum
+    std::vector<double> bins = {
+        5.0, 7.0, 10.0, 15.0, 20.0, 25.0, 30.0, 35.0, 40.0, 45.0, 50.0,
+        60.0, 70.0, 80.0, 90.0, 100.0
+    };
+
+    // ---
+    // Construct the objects that are needed to fill the grid
+
+    // choose the Evolution Basis to represent the grid
+    pineappl_pid_basis pid_basis = PINEAPPL_PID_BASIS_EVOL;
+    // define the types of hadrons and set them to be Unpolarised
+    pineappl_conv convs[] = {
+        {PINEAPPL_CONV_TYPE_UNPOL_PDF, 2212}, // proton
+        {PINEAPPL_CONV_TYPE_UNPOL_PDF, 2212}, // proton
+        {PINEAPPL_CONV_TYPE_UNPOL_FF, 211},   // pion
+    };
+
+    // define the kinematics object `kinematics = (μ, x1, x2, x)`
+    pineappl_kinematics scales = {PINEAPPL_KINEMATICS_SCALE, 0};
+    pineappl_kinematics x1 = {PINEAPPL_KINEMATICS_X, 0};
+    pineappl_kinematics x2 = {PINEAPPL_KINEMATICS_X, 1};
+    pineappl_kinematics z = {PINEAPPL_KINEMATICS_X, 2};
+    pineappl_kinematics kinematics[4] = {scales, x1, x2, z};
+
+    // define the specificities of the interpolations `interpolations = (μ, x1, x2, z)`
+    pineappl_reweight_meth scales_reweight = PINEAPPL_REWEIGHT_METH_NO_REWEIGHT;
+    pineappl_reweight_meth moment_reweight = PINEAPPL_REWEIGHT_METH_APPL_GRID_X;
+    pineappl_map scales_mapping = PINEAPPL_MAP_APPL_GRID_H0;
+    pineappl_map moment_mapping = PINEAPPL_MAP_APPL_GRID_F2;
+    pineappl_interp_meth interpolation_meth = PINEAPPL_INTERP_METH_LAGRANGE;
+    pineappl_interp interpolations[4] = {
+        {1e2, 1e8, 40, 3, scales_reweight, scales_mapping, interpolation_meth},  // μ
+        {2e-7, 1.0, 50, 3, moment_reweight, moment_mapping, interpolation_meth}, // x1
+        {2e-7, 1.0, 50, 3, moment_reweight, moment_mapping, interpolation_meth}, // x2
+        {2e-7, 1.0, 50, 3, moment_reweight, moment_mapping, interpolation_meth},  // z
+    };
+
+    // define the values of the unphysical scales `mu_scales = (μR, μF, μD)`
+    // where here we do not consider the fragmentation scale μD
+    pineappl_scale_func_form scale_mu = {PINEAPPL_SCALE_FUNC_FORM_SCALE, 0};
+    pineappl_scale_func_form mu_scales[3] = {scale_mu, scale_mu, scale_mu};
+
+    // ---
+    // Create the grid, fill it with Monte Carlo weights, and dump into disk
+
+    auto* grid = pineappl_grid_new2(bins.size() - 1, bins.data(), orders.size() / 5,
+        orders.data(), channels, pid_basis, convs, nb_convolutions + 1,
```

Comment:
Instead of `nb_convolutions + 1` you could write `sizeof (interpolations)`, which documents that this parameter just gives the length of the next array. This is also the safer choice, if someone decides to change the interpolation dimensions.